Client/mods/deathmatch/logic/lua/CLuaManager.h

```diff
@@ -52,6 +51,6 @@ class CLuaManager
     CRegisteredCommands* m_pRegisteredCommands;
 
     CFastHashMap<lua_State*, CLuaMain*> m_VirtualMachineMap;
-    std::list<CLuaMain*>                m_virtualMachines;
+    std::vector<CLuaMain*>              m_VirtualMachines;
```

Comment:
You are changing type from `std::list` to `std::vector`.

http://www.cplusplus.com/reference/vector/vector/erase/ says this:

> Because vectors use an array as their underlying storage, erasing elements in positions other than the vector end causes the container to relocate all the elements after the segment erased to their new positions. This is generally an inefficient operation compared to the one performed for the same operation by other kinds of sequence containers (such as list or forward_list).

Have you taken this into consideration?

Reply:
I did.
But, `std list` has a huge overhead on small datatypes, and heap allocates every.
While deliting from a list is `O(n)`, in reality that `O(n)` is far slower than vectors `O(n^2)`, because while std vector will easily be cached, iterating thru it is very fast, compared to std::list, thanks to the CPU cache.
Also, copying it is really fast as well, since we're talking about pointers, which are 4 bytes. Even in the worst case, this will take maybe n * 10 clock cycles to copy, which is very fast, compared to `std::list` where we need to wait a lot more.

Tldr, yes I did, and std::vector will always be faster (if `sizeof(T)` is small (like a pointer is 4 bytes))

Reply:
Okay 👍